hash-based-routing.html.md.erb

```diff
@@ -0,0 +1,47 @@
+---
+title: Hash-Based Routing
+owner: CF for VMs Networking
+---
+
+## <a id="purpose"></a> Purpose
+
+Hash-Based Routing is a load-balancing algorithm that distributes incoming requests to application instances based on a hash of a specific HTTP header value. This ensures consistent routing behavior where requests containing the same header value are always directed to the same instance.
+
+## <a id="key-features"></a> Key Features
+
+- **Configurable via Per-Route Options**: Easily set up Hash-Based Routing using application route options
+- **Consistent Hashing**: Implements the Maglev consistent hashing algorithm as outlined in the paper "Maglev: A Fast and Reliable Software Network Load Balancer" (https://storage.googleapis.com/gweb-research2023-media/pubtools/2904.pdf)
+- **Minimal Rehashing**: Uses the Maglev lookup table to map application instances by hash, minimizing the need to recalculate the entire table when instances are added or removed
+- **Configurable Hash Header**: Allows specifying which HTTP header to use for hashing via the `hash_header` property (e.g., `X-Resource-ID`)
+- **Sticky Session Precedence**: Prioritizes sticky sessions over hash-based routing when available
+- **Handling imbalanced loads**: Implements balance factor-based overload detection and mitigation
+- **No availability zones preference**: The global properties `locallyOptimistic` and `localAvailabilityZone` will be ignored when using Hash-Based Routing
+
+Hash-Based Routing implements a clear precedence hierarchy:
+
+1. **Sticky Sessions**: First checks if a sticky session is used and the sticky session endpoint is available
+2. **Hash-Based Routing**: Calculates the hash of the specified HTTP header value and routes the requests to a pre-determined application instance
+3. **Default Load Balancing**: Falls back to default load balancing (round-robin) if the header configured in the application route for hash-based routing is absent in the request
+
+
+## <a id="hash-based-vs-session-affinity"></a> Hash-Based Routing vs. Session Affinity
```

Comment:
above this is referenced as "sticky sessions", here "Session Affinity". Maybe we can unify?

Reply:
Tried to associate both terms and use _Session Affinity_ in most cases.